docs/learn/learn-xcvm.md

```diff
@@ -8,28 +8,108 @@ slug: ../learn-xcvm
+
+## XCMs are XCVM Programs
+
+A cross consensus message (XCM) is just a program that runs on the `XCVM`: in other words, one or
+more XCM instructions that are executed by the `xcm-executor`. To learn more about the XCVM and the
+XCM format, see the latest
+[blog post on XCM v3](https://medium.com/polkadot-network/xcm-part-iii-execution-and-error-management-ceb8155dd166)
 by Dr. Gavin Wood.
 
-XCM instructions might change a register, they might change the state of the consensus system or
-both.
+XCM instructions might change a register, the state of the consensus system, or both. Depending on
+the program's goal, whether it is to teleport assets from one chain to another or call a smart
+contract on another chain, XCMs usually require changes to the registers before any changes to the
+consensus system can be made.
+
+## XCM Executor & Configuration
+
+The XCM Executor's implementation centers around a core piece: the XCM configuration. Each instance
+of the Executor must have a valid configuration, which specifies a multitude of options on how a
+chain may treat incoming messages via
+[Barriers](https://github.com/paritytech/polkadot/blob/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor/src/config.rs#L52),
+calculate weight for a message via the
+[Weigher](https://github.com/paritytech/polkadot/blob/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor/src/config.rs#L55),
+how much weight to purchase via the
+[Trader](https://github.com/paritytech/polkadot/blob/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor/src/config.rs#L58),
+[configure fees](https://github.com/paritytech/polkadot/blob/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor/src/config.rs#L89),
+how to
+[convert origins](https://github.com/paritytech/polkadot/blob/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor/src/config.rs#L40),
+and more.
+
+## Cross Consensus Message (XCM) Anatomy & Flow
+
+An XCM is made up of a list of instructions that are executed in order. There are four different
+kinds of XCM instructions:
 
-One example of such an instruction would be `TransferAsset` which is used to transfer an asset to
-some other address on the remote system. It needs to be told which asset(s) to transfer and to
-whom/where the asset is to be transferred.
+1. **Instruction** - Results in a state change in the local consensus system or some state change.
+2. **Trusted Indication** - Tells the XCVM, or the Executor, that some action has been done before
+   already - meaning, this action is now trusted and can be acted on, i.e., in a teleport scenario.
+3. **Information** - Provides additional information about a particular origin, usually the result
+   of a query, i.e., a `QueryResponse` instruction.
+4. **System Notification** - Typically used in the context of when an HRMP channel is being opened,
+   closed, or accepted.
 
+Typically, an XCM takes the following path:
```

Comment:
This doesn't look like a series of steps an XCM goes through on the XCVM. It should be something like:
1. An XCM instruction is read
2. The instruction is executed. This means that the current values of the XCVM registers, the instruction type, and the instruction operands are all used to execute some operation, which might result in some registers changing their value, or in an error being thrown, which halts execution
3. The next instruction is read and so on until the end of the message

Reply:
I suppose it covers more than the XCVM - will modify as needed